File: admin-web/src/api/rateLimit.js
import service from '@/utils/request'

// createRateLimitRule 新增全局或指定接口限流规则。
export const createRateLimitRule = (data) => {
  return service({
    url: '/system/rate-limit/create',
    method: 'post',
    data
  })
}

// updateRateLimitRule 修改限流规则，后端保存后会立即刷新当前实例。
export const updateRateLimitRule = (data) => {
  return service({
    url: '/system/rate-limit/update',
    method: 'put',
    data
  })
}

// changeRateLimitRuleStatus 启用或禁用限流规则。
export const changeRateLimitRuleStatus = (data) => {
  return service({
    url: '/system/rate-limit/changeStatus',
    method: 'put',
    data
  })
}

// deleteRateLimitRule 删除指定限流规则。
export const deleteRateLimitRule = (params) => {
  return service({
    url: '/system/rate-limit/delete',
    method: 'delete',
    params
  })
}

// getRateLimitRuleDetail 查询限流规则详情。
export const getRateLimitRuleDetail = (params) => {
  return service({
    url: '/system/rate-limit/detail',
    method: 'get',
    params
  })
}

// getRateLimitRulePage 分页查询限流规则。
export const getRateLimitRulePage = (params) => {
  return service({
    url: '/system/rate-limit/pageList',
    method: 'get',
    params
  })
}

// refreshRateLimitRules 从 MySQL 重新加载启用规则到当前服务实例内存。
export const refreshRateLimitRules = () => {
  return service({
    url: '/system/rate-limit/refresh',
    method: 'post'
  })
}

File: server-api/internal/system/router/rate_limit.go
package router

import (
	handler "megin/internal/admin-api/system"
	"megin/pkg/context/router"
)

// RateLimitRouter 注册系统级 API 限流规则管理接口。
func RateLimitRouter(adminApiGroup *router.RouteGroup) *router.RouteGroup {
	rateLimit := &handler.RateLimitRule{}
	router.POST(adminApiGroup, "/system/rate-limit/create", rateLimit.Create)
	router.PUT(adminApiGroup, "/system/rate-limit/update", rateLimit.Update)
	router.PUT(adminApiGroup, "/system/rate-limit/changeStatus", rateLimit.ChangeStatus)
	router.DELETE(adminApiGroup, "/system/rate-limit/delete", rateLimit.Delete)
	router.GET(adminApiGroup, "/system/rate-limit/detail", rateLimit.Detail)
	router.GET(adminApiGroup, "/system/rate-limit/pageList", rateLimit.PageList)
	router.POST(adminApiGroup, "/system/rate-limit/refresh", rateLimit.Refresh)

	return adminApiGroup
}

// ScheduledTaskRouter 注册定时任务及执行日志管理接口。
func ScheduledTaskRouter(adminApiGroup *router.RouteGroup) *router.RouteGroup {
	task := &handler.ScheduledTask{}
	router.GET(adminApiGroup, "/system/scheduled-task/options", task.Options)
	router.POST(adminApiGroup, "/system/scheduled-task/create", task.Create)
	router.PUT(adminApiGroup, "/system/scheduled-task/update", task.Update)
	router.DELETE(adminApiGroup, "/system/scheduled-task/delete", task.Delete)
	router.GET(adminApiGroup, "/system/scheduled-task/detail", task.Detail)
	router.GET(adminApiGroup, "/system/scheduled-task/pageList", task.PageList)
	router.POST(adminApiGroup, "/system/scheduled-task/execute", task.Execute)
	router.GET(adminApiGroup, "/system/scheduled-task/logPageList", task.LogPageList)
	return adminApiGroup
}
